Add tailwind and CSS config to frontend

dcalacci opened the pull request:
This PR adds CSS configuration in the form of [tailwind](https://tailwindcss.com/) to the repo, including necessary CSS processing and other requirements.

- tailwind configuration at tailwind.config.js
- uses craco instead of react-create-app in order to alter postCSS
config
- updates nav bar styling using tailwind (to start)
- adds needed packages for tailwind in package.json and yarn.lock
- imports tailwind necessities in index.css

to do:
- [ ] update rest of frontend to use tailwind classes

dcalacci committed:
Adds tailwind + craco config

- tailwind configuration at tailwind.config.js
- uses craco instead of react-create-app in order to alter postCSS
config
- updates nav bar styling using tailwind (to start)
- adds needed packages for tailwind in package.json and yarn.lock
- imports tailwind necessities in index.css

dcalacci committed:
Adds tailwind + craco config

- tailwind configuration at tailwind.config.js
- uses craco instead of react-create-app in order to alter postCSS
config
- updates nav bar styling using tailwind (to start)
- adds needed packages for tailwind in package.json and yarn.lock
- imports tailwind necessities in index.css

aberke commented:
I <3 detailed commit messages

aberke commented on frontend/src/components/navigation.jsx:
```diff
@@ -1,54 +1,58 @@
-          </Link>
+      <nav className="navbar navbar-expand navbar-dark border-b border-yellow">
+        <div className="container w-full flex flex-row justify-between">
+          <div class="flex flex-row justify-self-start flex-grow justify-start">
```
react complains about 'class' vs 'className'. Is it a problem or something to ignore?

aberke commented on frontend/src/components/navigation.jsx:
```diff
@@ -1,54 +1,58 @@
+              </div>
+            </div>
+          </div>
+          <div class="justify-self-end font-black">
```
same as above

dcalacci replied:
react should use `className`, see https://stackoverflow.com/questions/46989454/class-vs-classname-in-react-16#46991278

Should fix!

aberke commented on frontend/package.json:
```diff
@@ -3,20 +3,24 @@
   "version": "0.1.0",
```
hey haven't been making any slick front-ends for the past 4 years. When I'm pulling in new changes to the package.json, should I run `yarn` ?

dcalacci replied:
yeah just run `yarn install`

aberke reviewed (approved):
Do you mind updating the README or creating a frontend README with your intended build/test/run flow? 
If it's just the same as before then no worries.

dcalacci commented:
To close: 

- [ ] React should use `className`, not `class`. see react should use className, see https://stackoverflow.com/questions/46989454/class-vs-classname-in-react-16#46991278
- [ ] Add tailwind to all other existing UI elements for consistency and ease of development

aberke committed:
updated poster link in README

aberke committed:
update to README on set up instructions

aberke committed:
initial models and tables with migration

- alembic is used for database management
- initial tables for simple models with one-to-many relationships set up

still todo:

- Tag table
- geocoordinates (lat,lng) in Locker model
- an API with tests

aberke committed:
Merge pull request #5 from aberke/v0-models

initial models and tables with migration

arunasank committed:
Make some formatting fixes

aberke committed:
Merge pull request #6 from aberke/break-ice

Make some formatting fixes

dcalacci committed:
Merge branch 'feature/add-tailwind' of github.com:aberke/locker-hack into feature/add-tailwind
